lib/ansible/modules/cloud/vmware/vmware_host_ntp.py

```diff
@@ -16,82 +16,126 @@
 DOCUMENTATION = r'''
 ---
 module: vmware_host_ntp
-short_description: Manage NTP configurations about an ESXi host
+short_description: Manage NTP server configuration of an ESXi host
 description:
-- This module can be used to manage NTP configuration information about an ESXi host.
+- This module can be used to configure, add or remove NTP servers from an ESXi host.
+- If C(state) is not given, the NTP servers will be configured in the exact sequence.
 - User can specify an ESXi hostname or Cluster name. In case of cluster name, all ESXi hosts are updated.
 version_added: '2.5'
 author:
 - Abhijeet Kasurde (@Akasurde)
+- Christian Kotte (@ckotte)
 notes:
 - Tested on vSphere 6.5
 requirements:
 - python >= 2.6
 - PyVmomi
 options:
-  cluster_name:
-    description:
-    - Name of the cluster.
-    - NTP settings are applied to every ESXi host system in the given cluster.
-    - If C(esxi_hostname) is not given, this parameter is required.
   esxi_hostname:
     description:
-    - ESXi hostname.
-    - NTP settings are applied to this ESXi host system.
-    - If C(cluster_name) is not given, this parameter is required.
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
   ntp_servers:
     description:
-    - "IP or FQDN of NTP server/s."
+    - "IP or FQDN of NTP server(s)."
     - This accepts a list of NTP servers. For multiple servers, please look at the examples.
+    type: dict
     required: True
   state:
     description:
-    - "present: Add NTP server/s, if it specified server/s are absent else do nothing."
-    - "absent: Remove NTP server/s, if specified server/s are present else do nothing."
-    default: present
+    - "present: Add NTP server(s), if it specified server(s) are absent else do nothing."
+    - "absent: Remove NTP server(s), if specified server(s) are present else do nothing."
     choices: [ present, absent ]
+    type: str
+  verbose:
+    description:
+    - Verbose output of the configuration change.
+    - Explains if an NTP server was added, removed, or if the NTP server sequence was changed.
+    type: bool
+    required: false
+    default: false
+    version_added: 2.8
 extends_documentation_fragment: vmware.documentation
 '''
 
 EXAMPLES = r'''
-- name: Set NTP setting for all ESXi Host in given Cluster
+- name: Configure NTP servers for an ESXi Host
   vmware_host_ntp:
-    hostname: '{{ vcenter_hostname }}'
-    username: '{{ vcenter_username }}'
-    password: '{{ vcenter_password }}'
-    cluster_name: cluster_name
+    hostname: vcenter01.example.local
```

Comment:
Do not change jinja variables to arbitrary values.

Reply:
Like I said before. Please discuss this with the core team (@dagwieers). There are different opinions on this matter. I agree with @dagwieers to use real values instead of Jinja variables..

Reply:
Hi
Maybe a bit of confusion here, hope I can resolve that

@Akasurde is part of the Ansible Core Team, and is responsible for the VMware modules
@dagwieers has commit, though isn't part of the Core Team

The comments that @Akasurde have made here need addressing (ie please revert the change to the credentials and cluster_name)

Thanks in advance,

Gundalow
Principal Engineer
Ansible by Red Hat

Reply:
Anyway. Discuss that internally. I'm tired of it...

Reply:
My advice was given when reviewing a new module. But I still stand by it.

Examples should be self-explanatory, and for that reason using real values is more meaningful than using '{{ cluster }}'. Using a Jinja variable as value does not give a clue to what the value really is, nor is it clear to the end-user where to define this (if he happens to know this is a variable). If it contains "cluster01" the user knows he needs to adapt it to a cluster name.

We simply cannot expect the reader to be an Ansible expert. Same for the credentials, which we hope to move to the connection-layer at some point in the future.

If this rationale is disputable, let's move this to the Documentation WG, but I hope it is self-evident that examples work better with real values.

Reply:
@ckotte I apologise that your PR has got caught up in discussions regarding how to improve the docs. For the moment please revert these changes and we can do a bulk tidy up once the new format has been decided, perhaps as part of https://github.com/ansible/ansible/issues/43433

We generally try and have these discussions outside of general PRs to avoid this exact back and forth.

For the record, I think examples using `example.com` and using documentation IP ranges are good. Though consistency in what we have is also important.